### ISSUE TITLE
Bug in Variation swtaches for jetwoobuilder product archive and product grid

### ISSUE THREAD

**Issue by sabbirahasanhabib** [CrocoBlock/suggestions#7227]
Hi, @UraraReika @Floydtm 

First of all take love from Bangladesh for Ukraine :)

I am using hello elementor child theme..My site is built on jetwoobuilder where i show products in archive pages(shop, category) by archive widgets...I also use product grid widget in the homepage, cart, and single product pages to show the best sell, favourite products, featured products etc...I wanted to add variation swatches so i bought Emran Ahmeds variation swatches pro and follow your guide from here>>

https://crocoblock.com/knowledge-base/tips-and-tricks/fail-to-add-variation-swatches-to-jetwoobuilder-products-grid-or-list/

### **Problem 1:** First i tried to implement swatches just only in archive pages so i put the code which was suggested at the last in the guide . Now the problem is >> as this code uses price hook it also puts swatches in the products grid widgets automatically at the top as same hook is present there also..I guess there has to be some exclusion for other pages except archive one's..

![Screenshot_51](https://github.com/Crocoblock/suggestions/assets/137334915/1a0b4c6f-0f6c-4ce7-a56f-68b60fff36ce)

### **problem 2:** I wanted to show the swatches in product grid widget also so i followed that guide carefully but the step 3(putting code to themes functions.php) breaks the entire site...then i use your repository code, although there was no site breaking but it shows no swatches in the frontend. I found there was a mismatch in the latest code of your repository and the guide in crocoblock official site. The mismatch is in this line 4>> function wvs_pro_archive_variation_button_args( $args, $product ) {....

**### So, In summary Archive shows extra swatches automatically in the product grid widget and the guide for the product grid widget doesnt work.**

**To Reproduce**
1. Plugins activated:
   - woocommerce
   - jetwoobuilder
   - elementor
   - elementor pro
   - variation swatches pro by emran ahmed
2. Theme: Hello child elementor
3. wordpress version: 6.4
4. php ver: 8.2

**Comment by UraraReika** [comment]
Problem 1: Our products loop widgets shares the same method of output products prices so in your case if you use Product Archive Item template only in the shop page you can use this snippet https://gist.github.com/Crocoblock/640ec669e84941d06ca1e360e56b3dc5#file-jet-woo-builder-variation-swatches-in-archive-template-php just add some condition wrapper like this 

```
add_filter( 'jet-woo-builder/template-functions/product-price', 'get_wvs_pro_archive_variation_template', 999, 1 );

function get_wvs_pro_archive_variation_template( $html ) {

	if ( is_shop() || is_product_taxonomy() || is_product_category() || is_product_tag() ) {
		$html .= wvs_pro_archive_variation_template();

		return '<div class="price">' . $html . '</div>';
	}
	
	return $html;

}
```

Problem 2: This snipped https://gist.github.com/Crocoblock/92698dccc093b92f37898c275f36fc2a#file-jet-woo-builder-ajax-add-to-cart-with-variation-swatches-php was forked so it a bit outdated even if it is not a very big deal use this one instead  - https://gist.github.com/UraraReika/24aaebe1bf340ca2c057ba64de2f1a87

**Comment by sabbirahasanhabib** [comment]
Thank you so much @UraraReika .. i have used the both code..it works like a charm..Amazing help buddy..but there is just one issue with the thumbnail effect of product grids widget which doesn't work...it supposed to show the second image on hover but it shows again the first image...thumbnail effect in product archive/loop works fine...Can you plz take a look on the codes again...

I tried to investigate what i found is>> adding this >>woo_variation_swatches()->show_archive_page_swatches();>> in the loop button breaks the thumb effect while having the code in the function.php

Thanks again and take love

**Comment by sabbirahasanhabib** [comment]
@UraraReika I am very sorry to tell you that problem 1 didn't solve...after updating jetwoobuilder to the latest version the codes showing no variation swatches...problem 2 is solved but only issue with thumbnail effect not working correctly..

I tried with this code by chatgpt which works:

```
add_filter( 'jet-woo-builder/template-functions/product-price', 'get_wvs_pro_archive_variation_template', 999, 1 );

function get_wvs_pro_archive_variation_template( $html ) {
    // Check if it's a WooCommerce product category, product tag, shop, or product taxonomy page
    if ( is_product_category() || is_product_tag() || is_shop() || is_tax() ) {
        // Additional logic related to archive variations
        $wvs_archive_variation_template = wvs_pro_archive_variation_template();
        $html .= $wvs_archive_variation_template;

        // Wrap the modified HTML in a <div> with class "price"
        return '<div class="price">' . $html . '</div>';
    }

    // If not on the specified pages, return the original HTML
    return $html;
}
```
But again thumbnail effect doesn't work

**Comment by UraraReika** [comment]
Latest update contains only small fixes and shouldn't broke swatches functionality. So in this case please [contact](https://support.crocoblock.com/support/home/ ) our support team.

For the thumbnail effect with swatches, originally this functionality has no compatibility with this swatches workaround but we will look close on this.

Thank you!

**Comment by sabbirahasanhabib** [comment]
Plz look at the thumbnail effect..The only issue is this now...Thanks again for your reply
